### ISSUE TITLE
`Conv2DTranspose` does not work right with `torch` backend 

### ISSUE THREAD

**Issue by suvadityamuk** [keras-team/keras-core#488]
I am trying to use `Conv2DTranspose` layer in a very simple and dumb example for a Segmentation network, with the Torch backend. In doing so, I am getting an error.

```python
Epoch 1/20
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
[<ipython-input-9-658c4c4b8133>](https://localhost:8080/#) in <cell line: 1>()
----> 1 history = model.fit(
      2     train_ds, epochs=20, validation_data=test_ds, callbacks=callbacks
      3 )

2 frames
[/usr/local/lib/python3.10/dist-packages/torch/nn/modules/module.py](https://localhost:8080/#) in _call_impl(self, *args, **kwargs)
   1499                 or _global_backward_pre_hooks or _global_backward_hooks
   1500                 or _global_forward_hooks or _global_forward_pre_hooks):
-> 1501             return forward_call(*args, **kwargs)
   1502         # Do not call functions when jit is used
   1503         full_backward_hooks, non_full_backward_hooks = [], []

RuntimeError: Exception encountered when calling Conv2DTranspose.call().

negative padding is not supported

Arguments received by Conv2DTranspose.call():
  • inputs=torch.Tensor(shape=torch.Size([16, 14, 14, 512]), dtype=float32)
```

[Reference Colab Notebook here](https://colab.research.google.com/drive/1L-5O1zf7MgvKJSchhbrVBXAU2dpdD-Kw?authuser=1#scrollTo=g8A7-1PmGHAT)

Please let me know how I can solve this issue.

**Comment by Sheiphan** [comment]
There is no code in the repo for `Conv2DTranspose` function. Only the function is created, but no logic has been written.

**Comment by fchollet** [comment]
Thanks for the report. Can you specify what arguments you called `Conv2DTranspose` with?

**Comment by suvadityamuk** [comment]
Sure, here you go

```python
x = keras.layers.Conv2DTranspose(filters=num_classes, kernel_size=7, strides=16, padding='same')(x)
```

*num_classes is 37

**Comment by fchollet** [comment]
This happens because of how we currently compute padding values in the case where the stride is larger than the input size. This line returns a negative value: https://github.com/keras-team/keras-core/blob/main/keras_core/backend/torch/nn.py#L471-L474

Will look into it.

**Comment by Frightera** [comment]
@fchollet Do you accept community contributions on that bug fixes right now? Or bug fixes are handled internally now?

**Comment by fchollet** [comment]
@Frightera absolutely -- if you can figure it out, please either open a PR or post your analysis in this thread 👍

**Comment by Frightera** [comment]
I am working on a fix. With the given parameters, padding becomes `(-4, -4)`.

**Comment by chenmoneygithub** [comment]
Thanks for reporting the issue! I made a fix in #520 since the conv transpose code in Torch backend is not quite humanly readable (thanks to the padding computation).
